src/app/extensions/order-templates/pages/account-order-template-detail/account-order-template-detail-line-item/account-order-template-detail-line-item.component.ts

```diff
@@ -0,0 +1,105 @@
+import { ChangeDetectionStrategy, Component, Input, OnChanges, OnInit, SimpleChanges } from '@angular/core';
+import { FormArray, FormControl, FormGroup } from '@angular/forms';
+import { Observable } from 'rxjs';
+
+import { ShoppingFacade } from 'ish-core/facades/shopping.facade';
+import { ProductView } from 'ish-core/models/product-view/product-view.model';
+import { ProductCompletenessLevel } from 'ish-core/models/product/product.model';
+
+import { OrderTemplatesFacade } from '../../../facades/order-templates.facade';
+import { OrderTemplate, OrderTemplateItem } from '../../../models/order-template/order-template.model';
+
+@Component({
+  selector: 'ish-account-order-template-detail-line-item',
+  templateUrl: './account-order-template-detail-line-item.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AccountOrderTemplateDetailLineItemComponent implements OnChanges, OnInit {
+  constructor(private productFacade: ShoppingFacade, private orderTemplatesFacade: OrderTemplatesFacade) {}
+
+  private static REQUIRED_COMPLETENESS_LEVEL = ProductCompletenessLevel.List;
+  @Input() orderTemplateItemData: OrderTemplateItem;
+  @Input() currentOrderTemplate: OrderTemplate;
+  @Input() selectedItemsForm: FormArray;
+
+  addToCartForm: FormGroup;
+  selectItemForm: FormGroup;
+  product$: Observable<ProductView>;
+
+  ngOnInit() {
+    this.initForm();
+    this.updateQuantities();
+  }
+
+  ngOnChanges(s: SimpleChanges) {
+    if (s.orderTemplateItemData) {
+      this.loadProductDetails();
+    }
+  }
+
+  updateQuantities() {
+    this.addToCartForm.valueChanges.subscribe(val =>
+      this.updateProductQuantity(this.orderTemplateItemData.sku, val.quantity)
+    );
+  }
+
+  /** init form in the beginning */
+  private initForm() {
+    this.addToCartForm = new FormGroup({
+      quantity: new FormControl(
+        this.orderTemplateItemData.desiredQuantity.value ? this.orderTemplateItemData.desiredQuantity.value : 1
+      ),
+    });
+
+    this.selectItemForm = new FormGroup({
+      productCheckbox: new FormControl(true),
+      sku: new FormControl(this.orderTemplateItemData.sku),
+    });
+
+    this.selectedItemsForm.push(this.selectItemForm);
+  }
+
+  addToCart(sku: string) {
```

Comment:
unused code, can be removed

Reply:
its being used in the html

Reply:
realized now in the commit that you mean addToCart, you are right :)